setup.py

```diff
@@ -317,11 +305,9 @@ def get_long_description():
             "flake8",
             "flake8-ets",
             "mypy",
-            # NumPy is not yet available for Python 3.12, but that should be
-            # fixed soon: https://github.com/numpy/numpy/issues/23808
-            "numpy; python_version < '3.12'",
+            "numpy",
             "pyface",
-            "PySide6; python_version >= '3.7' and python_version < '3.12'",
+            "PySide6; python_version < '3.12'",
```

Comment:
It's _possible_ that we don't need this restriction any more.

Reply:
The most recent release (6.6.2) claims compatibility with Python 3.12.  I wouldn't expect it to have support for 3.13.

Reply:
Yep, the tests seem fine with Python 3.12. I updated the restriction in [7c1306f](https://github.com/enthought/traits/pull/1773/commits/7c1306f181dbaa38ddc6e1d17631e212af147008)